### PR TITLE
Add example of how to create a Server Profile with OS Deployment Settings

### LINKED ISSUE #126: "Artifact Bundle with Server Profile" manifest
### Scenario/Intent
Adds a manifest to demonstrate how to create a server profile with a deployment plan.
The example must have the following tasks:

1. Uploads an artifact bundle zip file
2. Extracts the uploaded artifact bundle to create the artifacts on the appliance.
3. Ensure a Server Profile is present with the OS Deployment Plan, which was created in the previous step.

This example should demostrate how to provision an OS Image with a minimal OS Image that can be bootable.

### Environment Details
-**Puppet module for HPE OneView Version:** 2.2.0
-**OneView SDK Version:** 3.1.0
-**OneView Appliance Version:** 3.0
-**OneView Client API Version:** 300
-**Puppet Version:** 4.8
-**Ruby Version:** 2.2.6
-**Platform:** Windows/Linux

### Steps to Reproduce
Try to run the manifest "Artifact Bundle with Server Profile"

### Expected Result
Make the example manifest available.

### Actual Result
There is no example of how to create a server profile with a deployment plan.

### PR DESCRIPTION
**Pull request by marikrg**
### Description
Add an example of how to create a Server Profile with OS Deployment Settings

### Issues Resolved
#126 

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass (`$ rake test`).
- [x] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in the examples (please include helpful comments).
- [ ] Changes are documented in the CHANGELOG.

### CODE REVIEW

**Comment by fgbulsoni** [comment]
@marikrg Do we need to upload an artifact bundle here? Specially one so small? 
I'd rather have each artifact created by the manifest to highlight exactly what we are doing here. Could we do that?

**Comment by marikrg** [comment]
@fgbulsoni 
Not sure. I'm going to ask for the person who suggested this user case scenario.

**Comment by fgbulsoni** [comment]
OK, it was explained to me that this zip actually contains a super small bootable image along with a few resources created. 
An issue will be opened to better extend this user story having another example where each resource declaration, from the golden image upload to each artifact being created, is performed by a single manifest and that should give clarity to the final user.
This example will be considered as accepted for the case in which we have already downloaded an artifact bundle and want to perform a deploy from there, but only once we have the issue up on github.

File: examples/image_streamer/server_profile_deployment_settings.pp [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+# This example demonstrates how to create an advanced Gen 9 profile with OS Deployment Settings
+
```
Comment (fgbulsoni) [inline examples/image_streamer/server_profile_deployment_settings.pp]: Explain here what is contained inside the .zip and also better explain what the script will be doing.

**Review by fgbulsoni — CHANGES_REQUESTED** [review]

File: examples/image_streamer/server_profile_deployment_settings.pp (hunk 1 above)
Comment (marikrg) [reply examples/image_streamer/server_profile_deployment_settings.pp]: :+1:

**Comment by marikrg** [comment]
@fgbulsoni I realised that I must assign a server hardware to make this example more complete. I'm waiting for a Server Harware be available to make this change.

**Comment by marikrg** [comment]
> @fgbulsoni I realised that I must assign a server hardware to make this example more complete. I'm waiting for a Server Harware be available to make this change.

I have changed the Server Hardware affinity to Bay and Server and created a connection for management.

I still need to figure out why the added zip file is corrupted, I assume it's related with the eol config from .gitattributes.

Also pending create the issue that we've mentioned on this conversation.

**Comment by marikrg** [comment]
@fgbulsoni Done! Can you review it again?

File: examples/image_streamer/server_profile_deployment_settings.pp [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+################################################################################
+
+# This example demonstrates how to create an advanced Gen 9 profile with OS Deployment Settings.
+#
+# The Artifact Bundle Zip file contains a Golden Image, Build Plans, Plan Scripts, and a Deployment Plan.
+# The Deployment Plan will provision an OS Image that is not really an OS but rather a very minimal
+# OS Volume having only UEFI boot loader content in a UEFI Extended System Partition.
+# It has a simple configuration to show a message on the server console at boot.
+#
+# Once you run the manifest, you must:
```
Comment (fgbulsoni) [inline examples/image_streamer/server_profile_deployment_settings.pp]: Once this manifest is run, the following actions must be taken:

**Review by fgbulsoni — CHANGES_REQUESTED** [review]

**Review by fgbulsoni — APPROVED** [review]

**Pull request merged**